index.js

```diff
@@ -23,30 +26,31 @@ module.exports = function(options) {
 
   var includeRegExp = new RegExp(prefix + 'include\\s*\\([^)]*["\'](.*?)["\'](,\\s*({[\\s\\S]*?})){0,1}\\s*\\)+');
 
-  function fileInclude(file) {
-    var self = this;
-
+  function fileInclude(file, enc, cb) {
     if (file.isNull()) {
-      self.emit('data', file);
-    } else if (file.isStream()) {
+      cb(null, file);
+    }
+
+    if (file.isStream()) {
       file.contents.pipe(concat(function(data) {
         try {
-          self.emit('data', include(file, String(data)));
+          cb(null, include(file, String(data)));
         } catch (e) {
-          self.emit('error', new gutil.PluginError('gulp-file-include', e.message));
+          cb(new PluginError(PLUGIN_NAME, e.message));
         }
       }));
-    } else if (file.isBuffer()) {
+    }
+
+    if (file.isBuffer()) {
       try {
-        file = include(file, String(file.contents));
-        self.emit('data', file);
+        cb(null, include(file, String(file.contents)));
       } catch (e) {
-        self.emit('error', new gutil.PluginError('gulp-file-include', e.message));
+        cb(new PluginError(PLUGIN_NAME, e.message));
       }
     }
```

Comment:
No need to save this in variable for clusure, shorter and callback is pretty node pattern